packages/dm-core-plugins/src/table/types.ts

```diff
@@ -0,0 +1,47 @@
+import { IUIPlugin, TViewConfig } from '@development-framework/dm-core'
+
+export type TTablePlugin = {
+  editMode: boolean
+  columns: string[]
+  showDelete: boolean
+  editableColumns?: string[]
+  functionality: {
+    openAsTab: boolean
+    openAsExpandable: boolean
+    add: boolean
+    sort: boolean
+    edit: boolean
+    delete: boolean
+  }
+  defaultView: TViewConfig
+  views: TViewConfig[]
+}
+
+export const defaultConfig: TTablePlugin = {
+  editMode: true,
+  columns: ['name', 'type'],
+  showDelete: true,
+  editableColumns: [],
+  functionality: {
+    openAsTab: true,
+    openAsExpandable: false,
+    add: false,
+    sort: false,
+    edit: false,
+    delete: false,
+  },
+  defaultView: { type: 'ViewConfig', scope: 'self' },
+  views: [],
+}
+
+export type ITableItemRow = {
```

Comment:
```suggestion
export type TTableItemRow = {
```
prefix with T instead of I?